src/main/java/me/rigamortis/seppuku/impl/module/misc/HotBarRefillModule.java

```diff
@@ -0,0 +1,185 @@
+package me.rigamortis.seppuku.impl.module.misc;
+
+import me.rigamortis.seppuku.api.event.EventStageable;
+import me.rigamortis.seppuku.api.event.player.EventUpdateWalkingPlayer;
+import me.rigamortis.seppuku.api.module.Module;
+import me.rigamortis.seppuku.api.util.Timer;
+import me.rigamortis.seppuku.api.value.Value;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.client.gui.inventory.GuiInventory;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.ClickType;
+import net.minecraft.item.ItemStack;
+import team.stiff.pomelo.impl.annotated.handler.annotation.Listener;
+
+import java.util.ArrayList;
+
+/**
+ * Automatically refills the players hotbar.
+ *
+ * @author Old Chum
+ * @since 12/7/19
+ */
+public class HotBarRefillModule extends Module {
+    public final Value<Float> delay = new Value<>("Delay", new String[]{"Del"}, "The amount of delay in milliseconds.", 50.0f);
+    public final Value<Integer> refills = new Value<>("NumPerRefill", new String[]{"refill"}, "The number of total slots to be refilled at a time.", 2, 1, 10, 1);
+    public final Value<Boolean> offHand = new Value<>("OffHand", new String[]{"oh", "off", "hand"}, "If the off hand should be refilled.", true);
+
+    private Timer timer = new Timer();
+
+    public HotBarRefillModule() {
+        super("HotBarRefill", new String[]{"Replenish", "Refill", "AutoHotBar", "hbr"}, "NONE", -1, ModuleType.MISC);
+    }
+
+    @Listener
+    public void onWalkingUpdate (EventUpdateWalkingPlayer event) {
+        Minecraft mc = Minecraft.getMinecraft();
+
+        if (this.timer.passed(this.delay.getValue())) {
+            if (event.getStage() == EventStageable.EventStage.PRE) {
+                if (mc.currentScreen instanceof GuiInventory) {
+                    return;
+                }
```

Comment:
though this might interfere with things, I've seen other clients have settings to disable this check, should we do this here?

Reply:
I'm not sure what it will do with other clients, I took this check from [AutoArmor](https://github.com/seppukudevelopment/seppuku/blob/master/src/main/java/me/rigamortis/seppuku/impl/module/combat/AutoArmorModule.java#L38). If it does cause problems, I'm not sure what check to use in its place. Either way, it kinda sounds like a clippy moment to me.

Reply:
That's not what I meant; I meant that other clients have options for mods like this to toggle whether to disable the mod from functioning when the inventory is open.
Hence why I'm asking if we should do the same thing.

Reply:
nvm, was confused between this and having the mod **only** be enabled while the inventory is open